templates/server/src/main/resources/archetype-resources/api/src/test/java/__packageInPathFormat__/ToTest.java

```diff
@@ -55,12 +55,30 @@ public void testTos() {
     assertion.assertAll();
   }
 
+  /**
+   * Checks if there is an overridden version of the equals() and hashCode() method
+   */
+  private boolean doEqualsAndHashCodeExist(Class<?> clazz) {
+
+	boolean existing = false;
+
+	try {
+	  if(clazz.getMethod("equals", Object.class).getAnnotation(Override.class) != null) existing = true;
+	} catch(NoSuchMethodException noEquals) { existing = false; }
+
+	try {
+	  if(clazz.getMethod("hashCode", null).getAnnotation(Override.class) != null) existing = true;
+	} catch(NoSuchMethodException noHashCode) { existing = false; }
```

Comment:
1. equals and hashCode of class Object is always there. So you have to ask for it of the clazz object
2. don't check for the Override annotation. It's an optional annotation and therefore should not be used to determine the existence.
3. We should test for both methods separately if possible. Just in case someone is just implementing an equals function and not a hashcode.

Reply:
Number 2 and 3 are good suggestions, I didn't catch those mistakes yesterday. 

I'm not sure what you mean by number 1 though. The method checks for equals() and hashCode() of the clazz object. The 'Object.class' parameter only specifies the parameter of the equals() method, so that the method signatures can be compared.

Reply:
The new implementation checks for both methods seperately, but runs the test only if both methods were overridden. I'll look into a way to make the verifier only test one of each if not both methods were implemented.